include/kafka/Error.h

```diff
@@ -86,10 +106,23 @@ class Error
         return _rkError ? rd_kafka_error_is_retriable(_rkError.get()) : Optional<bool>{};
     }
 
+    /**
+     * Show whether the error is an abortable transaction error.
+     *
+     * Note:
+     * 1. Only valid for transactional API.
+     * 2. If `true`, the producer must call `abortTransaction` and start a new transaction with `beginTransaction` to proceed with transactions.
+     */
+    bool transactionRequiresAbort() const
+    {
+        return _rkError ? rd_kafka_error_txn_requires_abort(_rkError.get()) : false;
+    }
+
 private:
-    rd_kafka_error_shared_ptr _rkError;       // For error with rich info
-    rd_kafka_resp_err_t       _rkRespErr{};   // For error with a simple response code
-    Optional<std::string>     _message;       // For additional detailed message
+    rd_kafka_error_shared_ptr _rkError;     // For error with rich info
+    rd_kafka_resp_err_t       _respErr{};   // For error with a simple response code
+    Optional<std::string>     _message;     // Additional detailed message (if any)
+    Optional<bool>            _fatal;       // Fatal flag (if any)
```

Comment:
What is the benefit of Optional<bool> vs a bool with default value to false?

Reply:
Sometimes, we have no idea whether it's `fatal` or not, -- e.g, if the `Error` came from a `rd_kafka_resp_err_t` (which is just a simple error number), we just can't get more info. While on the other hand, if the `Error` came from a `rd_kafka_error_t` (which is a rich-info error type from librdkafka), we could be sure whether it's `fatal`(or `retriable`) or not.

Reply:
What is the difference if we just use a bool member like
bool _fatal = false;

Reply:
> What is the difference if we just use a bool member like
> bool _fatal = false;

If we do that, while the `Error` comes from `rd_kafka_resp_err_t` (instead of `d_kafka_error_t`), the `isFatal()` would always return `false` (but actually the failure could be `fatal`).

Reply:
But when the **_fatal** type is bool,  are we able to set fatal as true via the following constructor?

> Error(rd_kafka_resp_err_t respErr, std::string message, bool fatal)
>         : _respErr(respErr), _message(std::move(message)), _fatal(fatal) {}

To me, I don't quite get the necessity of std::optional<bool>. Do we actually want the 3 states of std::optional<bool>, that is, non-existing, existing and true, existing and false? Or a bool flag, with true and false is sufficient?

It's not a very big deal, but it's just weird that we have to de-reference when we call 
> *isFatal()

Reply:
> it's just weird that we have to de-reference when we call
> 
> > *isFatal()

I agree on that, -- not quite a good interface for users.
I really hope the `rd_kafka_resp_err_t`s would be replaced by `rd_kafka_error_t`s more and more in future, which help us to determine whether the error is `fatal` or `retriable`. E.g.
```
int rd_kafka_error_is_fatal (const rd_kafka_error_t *error);
int rd_kafka_error_is_retriable (const rd_kafka_error_t *error);
```
At the very beginning, I also considered whether could just make `isFatal()` return `false` for `rd_kafka_resp_err_t` case (since the `fatal` is really rare); but things could become much complicated for `isRetriable`, -- it's just hard to tell from the `rd_kafka_resp_err_t` (simple error type with numbers).